src/gui/lazyfilelist.h

```diff
@@ -0,0 +1,329 @@
+            } else {
+                if (strcmp(files[0].name, dir.fno.fname) != 0) {
+                    // i.e. we didn't get the same entry as the zeroth entry (which may occur when populating the window with non-null firstDirEntry)
+                    // Make place in the window by standard item rotation downwards (to the right)
+                    std::rotate(files.rbegin(), files.rbegin() + 1, std::make_reverse_iterator(i)); // resi i pripad, ze tech fajlu je zatim mene
+                    // Save the entry
+                    i->CopyFrom(dir.fno);
+                    if (filesInWindow < WINDOW_SIZE) {
+                        ++filesInWindow;
+                    }
+                }
+            }
+            ++totalFiles;
+        }
+    }
+
+    /// This is what a user does with the knob - moves the window by 1 item up or down if the cursor is at the top end.
+    /// It rotates the items downwards (to the right) and computes the new top missing entry
+    /// by iterating through the directory's content (which is reading one or more FAT sectors).
+    /// @return true if the window was actually moved (i.e. not at the very beginning of the dir)
+    bool MoveUp() {
+        if (windowStartingFrom < 0)
+            return false;
+
+        std::rotate(files.rbegin(), files.rbegin() + 1, files.rend());
+
+        if (windowStartingFrom == 0) {
+            // special case code - add a ".."
+            files[0].SetDirUp();
+            --windowStartingFrom;
+            return true;
+        }
+
+        F_DIR_RAII_Iterator dir(path);
+        // clear the item at the zeroth position
+        files[0].Clear();
+        while (dir.FindNext()) {
+            if (LessEF(files[0], dir.fno) && LessFE(dir.fno, files[1])) {
+                // to be inserted, the entry must be greater than zeroth entry AND less than the first entry
+                files[0].CopyFrom(dir.fno);
+            }
+        }
+        --windowStartingFrom;
+        return true;
+    }
+
+    /// A similar operation like MoveUp, but moves the window down by 1 item.
+    /// @return true if the window was actually moved
+    bool MoveDown() {
+        if (totalFiles <= windowStartingFrom + WINDOW_SIZE + 1) {
+            return false; // no more files
+        }
+        std::rotate(files.begin(), files.begin() + 1, files.end());
+
+        F_DIR_RAII_Iterator dir(path);
+        // prepare the last entry for comparison in the do-while cycle
+        files[WINDOW_SIZE - 1].MakeLastFName();
+        while (dir.FindNext()) {
+            if (LessFE(dir.fno, files[WINDOW_SIZE - 1]) && LessEF(files[WINDOW_SIZE - 2], dir.fno)) {
+                // to be inserted, the entry must be greater than the pre-last entry AND less than the last entry
+                files[WINDOW_SIZE - 1].CopyFrom(dir.fno);
+            }
+        }
+        ++windowStartingFrom;
+        return true;
+    }
+
+private:
+    struct Entry {
+        bool isFile;
+        char name[96];
```

Comment:
I think the maximum length is slightly higher. 

What about using the `_MAX_LFN` macro from `FatFS`?